06_predict_survival/nbconverted/plot_survival_pc_comparison.py

```diff
@@ -0,0 +1,459 @@
+    
+all_performance_df = pd.concat(all_performance_df)
+all_performance_df.training_data.replace(to_replace=training_data_map, inplace=True)
+all_performance_df.head(10)
+
+
+# In[16]:
+
+
+group_cancer_types = all_performance_df.groupby(['cancer_type']).count().n_dim
+max_count = group_cancer_types.max()
+valid_cancer_types = group_cancer_types[group_cancer_types == max_count].index
+print(valid_cancer_types)
+
+
+# In[17]:
+
+
+sns.set({'figure.figsize': (8, 6)})
+sns.set_style('whitegrid')
+
+sns.pointplot(data=all_performance_df, x='n_dim', y='delta_cindex', hue='training_data',
+              hue_order=training_data_map.values())
+plt.xlabel('Number of PCs')
+plt.ylabel('cindex(signal) - cindex(shuffled)')
+plt.title('Performance for varying PC count, averaged over cancer types')
+
+
+# In[18]:
+
+
+cancer_type_avg = (
+    all_performance_df[all_performance_df.cancer_type.isin(valid_cancer_types)]
+      .groupby('cancer_type')
+      .mean()
+).delta_cindex
+cancer_type_avg.sort_values(ascending=False).head(10)
+
+
+# In[19]:
+
+
+cancer_type_sd = all_performance_df.groupby('cancer_type').std().delta_cindex
+cancer_type_cv = cancer_type_avg / cancer_type_sd
+cancer_type_cv.sort_values(ascending=False).head(10)
+
+
+# In[20]:
+
+
+sns.set({'figure.figsize': (10, 5)})
+sns.set_style('whitegrid')
+
+sns.pointplot(data=all_performance_df[all_performance_df.cancer_type == 'pancancer'],
+              x='n_dim', y='delta_cindex', hue='training_data', 
+              hue_order=training_data_map.values())
+plt.xlabel('Number of PCs', size=14)
+plt.ylabel('cindex(signal) - cindex(shuffled)', size=14)
+plt.legend(title='Training data', fontsize=13, title_fontsize=13, loc='upper left', ncol=2)
+plt.title('Pan-cancer survival performance, all data types', size=14)
+plt.ylim(-0.05, 0.3)
+
+if SAVE_FIGS:
+    plt.savefig(images_dir / 'all_pancan_survival.svg', bbox_inches='tight')
+    plt.savefig(images_dir / 'all_pancan_survival.png',
+                dpi=300, bbox_inches='tight')
+
+
+# In[21]:
+
+
+sns.set({'figure.figsize': (28, 5)})
+sns.set_style('whitegrid')
+fig, axarr = plt.subplots(1, 5)
+
+cancer_type_cv = cancer_type_cv[cancer_type_cv.index != 'pancancer']
+for ix, cancer_type in enumerate(cancer_type_cv.sort_values(ascending=False).index[:5]):
+    ax = axarr[ix]
+    sns.pointplot(data=all_performance_df[all_performance_df.cancer_type == cancer_type],
+                  x='n_dim', y='delta_cindex', hue='training_data',
+                  hue_order=training_data_map.values(), ax=ax)
+    ax.set_xlabel('Number of PCs')
+    ax.set_ylabel('cindex(signal) - cindex(shuffled)')
+    ax.set_title('{} survival performance'.format(cancer_type))
+    ax.set_ylim(-0.05, 0.4)
+    if ix != 0:
+        ax.get_legend().remove()
+        
+if SAVE_FIGS:
+    plt.savefig(images_dir / 'all_top_cancers_survival.svg', bbox_inches='tight')
+    plt.savefig(images_dir / 'all_top_cancers_survival.png',
+                dpi=300, bbox_inches='tight')
+
```

Comment:
Similar to my previous comment, I don't have an intuition for why it would be decreasing here. Especially since in this plot you are looking at the difference with the shuffled labels.

Reply:
Yeah, I'm not entirely sure why performance seems to decrease with added PCs for some cancer types and not for others. It may have something to do with sample size - I know sarcoma and the kidney cancers tend to have fewer samples than LGG and (obviously) fewer than the pan-cancer models, so maybe the added features are causing feature selection challenges? Especially since the features here are PCs which aren't really sparse in the same sense that (for example) raw gene expression data or methylation data might be, so trying to select informative features using elastic net from a dense high-dimensional set of PCs may not be the best way to go about this.

I'll keep this in mind; maybe it would be interesting to try something similar with increasing numbers of raw features to see if the results follow the same trend.